### PR TITLE
feat: Add onclick to codemirror gutter markers

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [ ] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [ ] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [x] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

### CODE REVIEW

**Comment by jbrundage** [comment]
@GordonSmith please review

File: packages/codemirror/src/Editor.ts [hunk 1]
```diff
@@ -51,6 +51,7 @@ export class Editor extends HTMLWidget {
         this._codemirror.setGutterMarker(line, "CodeMirror-guttermarker", marker);
         marker.onmouseenter = onmouseenter;
         marker.onmouseleave = onmouseleave;
+        marker.onclick = onclick;
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts:54]: @jbrundage Does onClick receive any arguments?

**Review by GordonSmith — COMMENTED** [review]
Looks good - just one quick question.

File: packages/codemirror/src/Editor.ts (hunk 1 above)
Comment (jbrundage) [reply packages/codemirror/src/Editor.ts:54]: It does. The PointerEvent is the first argument.

File: packages/codemirror/src/Editor.ts [hunk 2]
```diff
@@ -33,7 +33,7 @@ export class Editor extends HTMLWidget {
         return gutters;
     }
 
-    addGutterMarker(lineNumber: number, commentText: string, backgroundColor: string = null, fontFamily: string = null, fontSize: string = null, onmouseenter = () => {}, onmouseleave = () => {}) {
+    addGutterMarker(lineNumber: number, commentText: string, backgroundColor: string = null, fontFamily: string = null, fontSize: string = null, onmouseenter = () => {}, onmouseleave = () => {}, onclick = () => {}) {
```
Comment (GordonSmith) [inline packages/codemirror/src/Editor.ts:36]: Type is incorrect for `onclick`

**Review by GordonSmith — CHANGES_REQUESTED** [review]

File: packages/codemirror/src/Editor.ts (hunk 2 above)
Comment (jbrundage) [reply packages/codemirror/src/Editor.ts]: @GordonSmith this is resolved

**Review by GordonSmith — APPROVED** [review]

**Pull request merged**